### PR TITLE
feat(upload): make global decisions configurable in all upload pages

### PR DESCRIPTION
**Pull request by shaheemazmalmmd**
## Description

Make global decision configurable. from all upload pages.

### Changes

* add a function to store values in report_info page;

## How to test

* Upload a package and do some clearing with global decisions.
* Upload a some other version of package again and see if there are decisions applied automatically.
* Upload a some other version of package by clicking "Apply global decisions for selected upload" and check the global decisions in your package.

### CODE REVIEW

File: src/lib/php/Dao/ClearingDao.php [hunk 1]
```diff
@@ -87,6 +88,13 @@ private function getRelevantDecisionsCte(ItemTreeBounds $itemTreeBounds, $groupI
 
     $globalScope = DecisionScopes::REPO;
     $localScope = DecisionScopes::ITEM;
+    
+    $applyGlobal = $this->uploadDao->getGlobalDecisionSettingsFromInfo($uploadId);
+    if (!empty($applyGlobal)) {
+      $applyGlobal = "(ut.pfile_fk = cd.pfile_fk AND cd.scope = $globalScope) OR";
+    } else {
+      $applyGlobal = "";
```
Comment (GMishx) [inline src/lib/php/Dao/ClearingDao.php:96]: Unused variable. Should it be placed in the return sql string?

File: src/lib/php/UI/template/include/upload.html.twig [hunk 2]
```diff
@@ -37,6 +37,13 @@
           <input type="text" class="form-control" style="width:40%;" name="{{ descriptionInputName }}" value="{{ descriptionInputValue }}">
         </div>
       </li>
+      <li>
+        <div class="form-group">
+          <input type="checkbox" class="browse-upload-checkbox view-license-rc-size" name="globalDecisions" value="1"/>
+          {{ 'Apply global decisions for selected upload'| trans }}
+          <img src="images/info_16.png" data-toggle="tooltip" title="{{'Apply results of concluded files with global from previous uploads'|trans}}" alt="" class="info-bullet"/>
```
Comment (GMishx) [inline src/lib/php/UI/template/include/upload.html.twig:44]:
```suggestion
          <img src="images/info_16.png" data-toggle="tooltip" title="{{'Apply results of concluded files marked as global from previous uploads'|trans}}" alt="" class="info-bullet"/>
```

File: src/lib/php/UI/template/include/upload.html.twig [hunk 3]
```diff
@@ -37,6 +37,13 @@
           <input type="text" class="form-control" style="width:40%;" name="{{ descriptionInputName }}" value="{{ descriptionInputValue }}">
         </div>
       </li>
+      <li>
+        <div class="form-group">
+          <input type="checkbox" class="browse-upload-checkbox view-license-rc-size" name="globalDecisions" value="1"/>
+          {{ 'Apply global decisions for selected upload'| trans }}
```
Comment (GMishx) [inline src/lib/php/UI/template/include/upload.html.twig:43]:
```suggestion
          {{ 'Apply global decisions for current upload'| trans }}
```

File: src/www/ui/api/documentation/openapi.yaml [hunk 4]
```diff
@@ -435,6 +435,14 @@ paths:
               - protected
               - public
             default: protected
+        - name: applyglobal
```
Comment (GMishx) [inline src/www/ui/api/documentation/openapi.yaml:438]:
```suggestion
        - name: applyGlobal
```

File: src/www/ui/api/documentation/openapi.yaml [hunk 5]
```diff
@@ -435,6 +435,14 @@ paths:
...
+          description: >
+            Apply global decisions for selected upload
```
Comment (GMishx) [inline src/www/ui/api/documentation/openapi.yaml:440]:
```suggestion
            Apply global decisions for current upload
```

File: src/www/ui/api/documentation/openapi.yaml (hunk 4 above)
Comment (GMishx) [reply src/www/ui/api/documentation/openapi.yaml:438]: Please change the patch version of REST API to mark new changes.

File: src/www/ui/core-auth.php [hunk 6]
```diff
@@ -175,6 +175,7 @@ function updateSession($userRow)
     $SysConf['auth'][Auth::GROUP_ID] = $userRow['group_fk'];
     $this->session->set(Auth::GROUP_ID, $userRow['group_fk']);
     $_SESSION['GroupName'] = $userRow['group_name'];
+    $_SESSION['BannerOption'] = 1;
```
Comment (GMishx) [inline src/www/ui/core-auth.php:178]: Potential changes from other branch?

File: src/www/ui/core-schema.dat [hunk 7]
```diff
@@ -2107,6 +2107,8 @@
   $Schema["TABLE"]["report_info"]["ri_unifiedcolumns"]["ADD"] = "ALTER TABLE \"report_info\" ADD COLUMN \"ri_unifiedcolumns\" json";
   $Schema["TABLE"]["report_info"]["ri_unifiedcolumns"]["ALTER"] = "ALTER TABLE \"report_info\" ALTER COLUMN \"ri_unifiedcolumns\" DROP NOT NULL";
 
+  $Schema["TABLE"]["report_info"]["ri_globaldecision"]["DESC"] = "";
+  $Schema["TABLE"]["report_info"]["ri_globaldecision"]["ADD"] = "ALTER TABLE \"report_info\" ADD COLUMN \"ri_globaldecision\" int2 DEFAULT 0";
```
Comment (GMishx) [inline src/www/ui/core-schema.dat:2112]: Please add missing `ALTER` statement for the new column.

**Review by GMishx — CHANGES_REQUESTED** [review]
Some suggestions about changes.

**Comment by GMishx** [comment]
There seems to be some inconsistency with the results.
I had an upload with GPL-3.0 license where global decision of GPL-3.0 was added. After re-uploading the same package without selecting "Apply global decisions for current upload", the file appears cleared in tree view.

image

And single file view shows file as cleared.

image

But Unified Clearing report has no license decision.

**Review by ag4ums — APPROVED** [review]
Code looks good and work as expected

**Pull request merged**